libs/editor/egui_editor/src/layout_job.rs

```diff
@@ -28,127 +30,122 @@ pub enum Annotation {
+        if style.elements.contains(&Element::Item) {
+            let indent_level = style
                 .elements
                 .iter()
-                .any(|el| matches!(el, Element::Heading(_)))
-            {
-                ret.annotation = Some(Annotation::Rule);
-            }
-        }
-
-        ret.append(src, vis, region, absorb_terminal_nl);
-        ret
-    }
-
-    pub fn append(
-        &mut self, src: &str, vis: &VisualAppearance, data: &StyledChunk, absorb_terminal_nl: bool,
-    ) {
-        let text_format = data.text_format(vis);
-
-        self.range.start = min(self.range.start, data.range.start);
-        self.range.end = max(self.range.end, data.range.end);
-
-        let mut string = &src[data.range.start.0..data.range.end.0];
-        let mut range = data.range.clone();
-
-        if let Some(Item(Bulleted, _)) = self.annotation {
-            if data.block_start {
-                let t = string.trim_start();
-                self.head_modification = string.len() - t.len();
-                if t.starts_with("+ ") || t.starts_with("* ") || t.starts_with("- ") {
-                    self.head_modification += 2;
-                } else if t.starts_with('+') || t.starts_with('*') || t.starts_with('-') {
-                    self.head_modification += 1;
+                .filter(|&e| e == &Element::Item)
+                .count() as IndentLevel;
+            let text = {
+                let trimmed_text = text.trim_start();
+                head_size = text.len() - trimmed_text.len();
+                trimmed_text
+            };
+            if text.starts_with("+ ") || text.starts_with("* ") || text.starts_with("- ") {
+                annotation = Some(Annotation::Item(ItemType::Bulleted, indent_level));
+                head_size += 2;
+            } else if let Some(prefix) = text.split(". ").next() {
+                if let Ok(num) = prefix.parse::<usize>() {
+                    annotation = Some(Annotation::Item(ItemType::Numbered(num), indent_level));
+                    head_size += prefix.len() + 2;
                 }
-
-                range.start += self.head_modification;
             }
+        } else if style.elements.contains(&Element::Heading(HeadingLevel::H1)) {
+            annotation = Some(Annotation::Rule);
         }
 
-        if absorb_terminal_nl && string.ends_with('\n') {
-            self.tail_modification = 1;
-        }
-        range.end -= self.tail_modification;
+        (annotation, head_size, Self::tail_size(style, src, absorb_terminal_nl))
+    }
 
-        string = &src[range.start.0..range.end.0];
-        self.job.append(string, 0.0, text_format);
+    fn tail_size(style: &StyleInfo, src: &str, absorb_terminal_nl: bool) -> usize {
+        usize::from(
+            style.range.end.0 > 0
+                && absorb_terminal_nl
+                && &src[style.range.end.0 - 1..style.range.end.0] == "\n",
+        )
     }
 }
 
-impl Editor {
```

Comment:
basically unchanged; de-indenting and re-wrapping the text here makes it look like more than it is